fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -828,7 +828,10 @@ void assertResolutionStateMutationsSizeConsistent(const std::vector<ResolveTrans
 
 	for (int r = 1; r < resolution.size(); r++) {
 		ASSERT(resolution[r].stateMutations.size() == resolution[0].stateMutations.size());
-		ASSERT_EQ(resolution[0].privateMutationCount, resolution[r].privateMutationCount);
+		// The resolution[r].privateMutationCount can be different due to random log router tags:
+		// two consecutive mutations may have the same or different locations. If they share the
+		// same locations, then no new SpanContextMessage is written. Otherwise, a new one is written.
```

Comment:
I don't fully follow this, but that's fine. Maybe you can say, at a high level, what a SpanContextMessage is about?

Reply:
`SpanContextMessage` is a debug ID attached to a transaction and is inserted as a mutation. So for two normal mutations, we could have two sequence:

- `SpanContextMessage mutation1 mutation2` if they share the same destination; or
- `SpanContextMessage mutation1 SpanContextMessage mutation2` if they have different destinations (caused by random log router tag);

Thus the mutation count can be different.

Reply:
Ok, thanks for the explanation!